src/main/java/de/zalando/zmon/dataservice/proxies/checks/DefaultChecksService.java

```diff
@@ -17,18 +21,31 @@ public DefaultChecksService(DataServiceConfigProperties config) {
 		this.config = config;
 	}
 
+	public static HttpClient getHttpClient(int socketTimeout, int timeout, int maxConnections) {
+		RequestConfig config = RequestConfig.custom().setSocketTimeout(socketTimeout).setConnectTimeout(timeout).build();
+		return HttpClients.custom().setMaxConnPerRoute(maxConnections).setMaxConnTotal(maxConnections).setDefaultRequestConfig(config).build();
+	}
+
 	@Override
 	public String allActiveAlertDefinitions(String query) throws URISyntaxException, IOException {
+
+		final Executor executor = Executor.newInstance(getHttpClient(100, 5000, 1));
```

Comment:
100ms socket timeout? that will probably not work, we also need to make these settings configurable (Spring properties class)